### PR TITLE
CLN: Disallow use of null/None for alias/stratigraphic_alias

### PR DESCRIPTION
**Pull request by janbjorge**
Addresses: https://github.com/equinor/fmu-dataio/issues/373

### CODE REVIEW

**Comment by perolavsvendsen** [comment]
This means that if e.g. `alias` in the global config (erroneously) is a `None` it will now raise?

I think it depends a bit on current behavior how this will impact:
- If current behavior is to crash violently with an obscure message, then this is a clear improvement.
- If current behavior is to raise with an informative message, this is also fine.
- If current behavior is to not raise, or handle this silently, this may be disruptive as some workflows that now work, will seize to work. If this is the case, we should probably go through a user warning for a while and clean the incoming information from the global config.

**Comment by janbjorge** [comment]
> This means that if e.g. `alias` in the global config (erroneously) is a `None` it will now raise?
> 
> I think it depends a bit on current behavior how this will impact:
> 
>     * If current behavior is to crash violently with an obscure message, then this is a clear improvement.
> 
>     * If current behavior is to raise with an informative message, this is also fine.
> 
>     * If current behavior is to not raise, or handle this silently, this may be disruptive as some workflows that now work, will seize to work. If this is the case, we should probably go through a user warning for a while and clean the incoming information from the global config.

If the *value* of _alias_ is `None` pydantic raises. If its lacking/unset there is a default-factory that will produce a empty-array/list. Any contents of the list must be of type string, meaning numbers can be represented as strings.

The current code (linked below) does not check if the value of "alias" is a list, so if the users gives anything other then a list it will break at this point. 

It might be an `AttributeError: 'NoneType' object has no attribute 'append'`, but if the user decides to create a new key-value mapping we would get an `AttributeError: 'dict' object has no attribute 'append'`. Looping back to the first statement, pydantic will raise if the contents of the "alias" key anything but empty or a list of strings.

https://github.com/janbjorge/fmu-dataio/blob/main/src/fmu/dataio/_objectdata_provider.py#L175

**Comment by perolavsvendsen** [comment]
> > This means that if e.g. `alias` in the global config (erroneously) is a `None` it will now raise?
> > I think it depends a bit on current behavior how this will impact:
> > ```
> > * If current behavior is to crash violently with an obscure message, then this is a clear improvement.
> > 
> > * If current behavior is to raise with an informative message, this is also fine.
> > 
> > * If current behavior is to not raise, or handle this silently, this may be disruptive as some workflows that now work, will seize to work. If this is the case, we should probably go through a user warning for a while and clean the incoming information from the global config.
> > ```
> 
> If the _value_ of _alias_ is `None` pydantic raises. If its lacking/unset there is a default-factory that will produce a empty-array/list. Any contents of the list must be of type string, meaning numbers can be represented as strings.
> 
> The current code (linked below) does not check if the value of "alias" is a list, so if the users gives anything other then a list it will break at this point.
> 
> It might be an `AttributeError: 'NoneType' object has no attribute 'append'`, but if the user decides to create a new key-value mapping we would get an `AttributeError: 'dict' object has no attribute 'append'`. Looping back to the first statement, pydantic will raise if the contents of the "alias" key anything but empty or a list of strings.
> 
> https://github.com/janbjorge/fmu-dataio/blob/main/src/fmu/dataio/_objectdata_provider.py#L175

OK 👍 

@rwiker will Sumo be affected with empty lists in produced metadata from FMU?

**Comment by rwiker** [comment]
Empty lists should be fine; it should be easy to check.

**Review by rwiker — APPROVED** [review]
LGTMBWDIK.

**Comment by perolavsvendsen** [comment]
> LGTMBWDIK.

...if Sumo handles empty lists when they start arriving.

**Comment by perolavsvendsen** [comment]
Main change here, as I see it, is that `fmu-dataio` will produce data where `alias` and `stratigraphic_alias` is present as an empty list if no values. Previously, it would be populated list or nothing at all (tag not present).

Will empty lists validate against the legacy schema? It is not discontinued yet, so when this hits `main` it also hits Komodo bleeding, and then it hits the nightly tests.

In short: We should probably allow empty lists in the legacy schema if they don't currently validate.

**Comment by perolavsvendsen** [comment]
@rwiker will Sumo handle empty lists? And is this allowed by the legacy schema?

If no data present:
Today: `stratigraphic_alias` will not be present at all in the metadata.
After this PR (if I understand this properly): `stratigraphic_alias` will be present, but as an empty list.

Current schema defines `stratigraphic_alias` as a list where each item is an instance of `data/name`, which is a `string`. So before we merge, the following should be checked:

- [ ] Current schema allows the data as they are produced post this PR
- [ ] Sumo accepts empty lists, also in indexed fields

**Comment by rwiker** [comment]
I just tested with one surface object, which had a single value in the list `data.alias`. If I remove the value, leaving an empty list, I’m still able to upload the document with the current json-schema.

Note that in elasticsearch, every attribute/property/tag/whatever can accept either no values at all (i.e, being left out), a single value, or a list of values. So, for data.alias, the following are all valid options

- (nothing)
- "alias": []
- "alias": "Therys Fm."
- "alias": ["Therys Fm."]
- "alias": ["Therys Fm.", "Formation Therys"]

**Comment by janbjorge** [comment]
Parking for now, we have other things to address.
